src/jaxsim/api/frame.py

```diff
@@ -0,0 +1,196 @@
+import functools
+from typing import Sequence
+
+import jax
+import jax.numpy as jnp
+import jaxlie
+import numpy as np
+
+import jaxsim.api as js
+import jaxsim.typing as jtp
+
+from .common import VelRepr
+
+# =======================
+# Index-related functions
+# =======================
+
+
+@functools.partial(jax.jit, static_argnames="frame_name")
+def name_to_idx(model: js.model.JaxSimModel, *, frame_name: str) -> jtp.Int:
+    """
+    Convert the name of a frame to its index.
+
+    Args:
+        model: The model to consider.
+        frame_name: The name of the frame.
+
+    Returns:
+        The index of the frame.
+    """
+
+    frame_names = np.array([frame.name for frame in model.description.get().frames])
+
+    if frame_name in frame_names:
+        idx_in_list = np.argwhere(frame_names == frame_name)
+        return jnp.array(idx_in_list + model.number_of_links()).squeeze().astype(int)
+    return jnp.array(-1).astype(int)
```

Comment:
Just an early comment, can we also handle the case in which the `frame_name` passed is not part of the model?

Reply:
do you mean if we check that frame_name exists in the frames of the model? because in this case we return `jnp.array(-1).astype(int)` as it is done for the corresponding link function https://github.com/xela-95/jaxsim/blob/96c600dd2e7fc9a0136800924fde8bbe40d9f5cd/src/jaxsim/api/link.py#L41-L42

Reply:
Quick comment: reporting errors of wrong indices in JIT compiled functions is not straightforward. When I started developing the functional APIs, I started exploring exception raised in compiled functions, but I was not happy with the first rather complex solution I found. For the time being, returning `-1` is good enough. I still have to wrap my head around reporting errors from compiled functions.

All of this is true when a compiled function accepts a link/joint/frame index. When a function, instead, accepts a static string as in this case, we can just `raise` a `ValueError`. Since it's something that should be propagated also to joint and links, I propose to do this for all of them in a new PR, maybe together with other types of error reporting.

Reply:
Alright, thanks for the explanations